### PR TITLE
Add pull "merged" notification subject status to API

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

File: modules/convert/notification.go [hunk 1]
```diff
@@ -47,6 +47,13 @@ func ToNotificationThread(n *models.Notification) *api.NotificationThread {
 			if err == nil && comment != nil {
 				result.Subject.LatestCommentURL = comment.APIURL()
 			}
+
+			pr, _ := n.Issue.GetPullRequest()
```
Comment (lunny) [inline modules/convert/notification.go:51]: Is there a pull.State() or should we create one?

Comment (6543) [reply modules/convert/notification.go:51]: there is a pull.State:
```go
	PullRequestStatusConflict PullRequestStatus = iota
	PullRequestStatusChecking
	PullRequestStatusMergeable
	PullRequestStatusManuallyMerged
	PullRequestStatusError
	PullRequestStatusEmpty
```
but this is not that helpfull info, if you like to build a UI via API showing notifications of issues/pulls

**Review by zeripath — APPROVED** [review]

File: modules/convert/notification.go [hunk 2]
```diff
@@ -47,6 +47,13 @@ func ToNotificationThread(n *models.Notification) *api.NotificationThread {
...
+			if pr != nil {
+				if pr.HasMerged {
+					result.Subject.State = "merged"
+				}
+			}
```
Comment (a1012112796) [inline modules/convert/notification.go]:
```suggestion
			if pr != nil && pr.HasMerged {
				result.Subject.State = "merged"
			}
```

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Pull request merged**